TYPO3.Flow/Classes/TYPO3/Flow/Cache/Backend/SimpleFileBackend.php

```diff
@@ -260,14 +255,23 @@ public function remove($entryIdentifier)
 
         $pathAndFilename = $this->cacheDirectory . $entryIdentifier . $this->cacheEntryFileExtension;
 
-        try {
-            $lock = new Lock($pathAndFilename);
-            unlink($pathAndFilename);
-            $lock->release();
-        } catch (\Exception $exception) {
+        if (strncasecmp(PHP_OS, 'WIN', 3) == 0) {
+            // On Windows, unlinking a locked/opened file will not work, so we just attempt the delete straight away.
+            // In the worst case, the unlink will just fail due to concurrent access and the caller needs to deal with that.
+            return @unlink($pathAndFilename);
+        }
+
+        $file = @fopen($pathAndFilename, 'r');
+        if ($file === false) {
```

Comment:
Good call.

So what is the intended/expected API here on file opening failure? I'd say for the remove operation just returning false and ignoring anything else is fine (as before). But for the reading/writing part it would make sense to somehow bubble exceptions up... actually there was no `try / catch` before, so it would maybe even make sense to throw exceptions on lock failure, too.